docker-app/qfieldcloud/core/tests/test_team.py

```diff
@@ -0,0 +1,410 @@
+import logging
+from qfieldcloud.authentication.models import AuthToken
+from qfieldcloud.core.models import (
+    TeamMember,
+    Organization,
+    OrganizationMember,
+    Person,
+    Team,
+)
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from .utils import setup_subscription_plans
+
+logging.disable(logging.CRITICAL)
+
+
+class TeamsTestCase(APITestCase):
+    def setUp(self):
+        setup_subscription_plans()
+
+        self.u1 = Person.objects.create_user(username="user1", password="abc123")
+        self.token1 = AuthToken.objects.get_or_create(user=self.u1)[0]
+
+        self.u2 = Person.objects.create_user(username="user2", password="abc123")
+        self.token2 = AuthToken.objects.get_or_create(user=self.u2)[0]
+
+        self.u3 = Person.objects.create_user(
+            username="user3", email="user3@email.com", password="abc123"
+        )
+        self.token3 = AuthToken.objects.get_or_create(user=self.u3)[0]
+
+        self.o1 = Organization.objects.create(
+            username="organization1",
+            password="abc123",
+            type=Organization.Type.ORGANIZATION,
+            organization_owner=self.u1,
+        )
+
+        OrganizationMember.objects.create(
+            organization=self.o1,
+            member=self.u2,
+            role=OrganizationMember.Roles.ADMIN,
+        )
+        OrganizationMember.objects.create(
+            organization=self.o1,
+            member=self.u3,
+            role=OrganizationMember.Roles.MEMBER,
+        )
+
+        team1_username = Team.format_team_name(self.o1.username, "team1")
+        team2_username = Team.format_team_name(self.o1.username, "team2")
+
+        self.t1 = Team.objects.create(
+            username=team1_username,
+            team_organization=self.o1,
+        )
+        self.t2 = Team.objects.create(
+            username=team2_username,
+            team_organization=self.o1,
+        )
+
+    def test_authorized_user_can_list_teams(self):
+        """Test that an authorized user can list teams."""
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.token1.key)
+
+        response = self.client.get(f"/api/v1/organizations/{self.o1.username}/teams/")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        payload = response.json()
+        team_usernames = sorted([team["username"] for team in payload])
+
+        self.assertEqual(len(team_usernames), 2)
+        self.assertEqual(self.t1.username, team_usernames[0])
+        self.assertEqual(self.t2.username, team_usernames[1])
```

Comment:
I see that down there in `test_non_admin_user_can_list_team_members` and other you use "assertJsonEqual". It is fine to use it, but here you properly catch the ordering that is not static. Let's assume the ordering will be fine for now and change the code to use `assertJsonEqual`.

What you can do is just sort the JSON properly:

```
teams = sorted(response.json(), key=lambda t: t["username"])

self.assertListEqual(teams, [
  {
      "username": self.t1.username,
  },
  {
      "username": self.t2.username,
  },
])
```

Reply:
Does not find assertJsonEqual in APITestCase class. Shall i try to inherit from the TestCase?